### PR TITLE
Fix for protocol upper/lower case handling

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Now lower case configured required and vCD returns lower case and they match

### CODE REVIEW

File: website/docs/r/dnat.html.markdown [hunk 1]
```diff
@@ -52,9 +52,9 @@ The following arguments are supported:
 * `port` - (Required) The port number to map. -1 translates to "any"
 * `translated_port` - (Optional) The port number to map
 * `internal_ip` - (Required) The IP of the VM to map to
-* `protocol` - (Optional; *v2.0+*) The protocol type. Possible values are TCP, UDP, TCPUDP, ICMP, ANY. TCP is default to be backward compatible with previous version
+* `protocol` - (Optional; *v2.0+*) The protocol type. Possible values are tcp, udp, tcpupd, icmp, any. `tcp` is default to be backward compatible with previous version
```
Comment (lvirbalas) [inline website/docs/r/dnat.html.markdown]: Please quote all these values. Now that they are lower case they need emphasis.

Comment (vbauzysvmware) [reply website/docs/r/dnat.html.markdown]: done

File: website/docs/r/dnat.html.markdown [hunk 2]
```diff
@@ -52,9 +52,9 @@ The following arguments are supported:
...
 * `translated_port` - (Optional) The port number to map
 * `internal_ip` - (Required) The IP of the VM to map to
-* `protocol` - (Optional; *v2.0+*) The protocol type. Possible values are TCP, UDP, TCPUDP, ICMP, ANY. TCP is default to be backward compatible with previous version
+* `protocol` - (Optional; *v2.0+*) The protocol type. Possible values are `tcp`, `udp`, `tcpupd`, `icmp`, `any`. `tcp` is default to be backward compatible with previous version
 * `icmp_sub_type` - (Optional; *v2.0+*) The name of ICMP type. Possible values are   address-mask-request, destination-unreachable, echo-request, echo-reply, parameter-problem, redirect, router-advertisement, router-solicitation, source-quench, time-exceeded, timestamp-request, timestamp-reply, any
-* `network_type` - (Optional; *v2.4+*) Type of the network on which to apply the NAT rule. Possible values org or ext. *`network_type` will be a required field in the next major version.*
+* `network_type` - (Optional; *v2.4+*) Type of the network on which to apply the NAT rule. Possible values `org` or `ext`. *`network_type` will be a required field in the next major version.*
```
Comment (lvirbalas) [inline website/docs/r/dnat.html.markdown]: Please do the same quoting in SNAT doc.

**Comment by lvirbalas** [comment]
BTW, in the warning of SNAT and DNAT pages:
```
Warning: When advanced edge gateway is used and the rule is updated using UI, then Id mapping will be lost and terraform won't find the rule anymore and remove it from state.
```
Please change:
1. "Id" -> "ID"
2. "terraform" -> "Terraform"

**Comment by vbauzysvmware** [comment]
> BTW, in the warning of SNAT and DNAT pages:
> 
> ```
> Warning: When advanced edge gateway is used and the rule is updated using UI, then Id mapping will be lost and terraform won't find the rule anymore and remove it from state.
> ```
> 
> Please change:
> 
>     1. "Id" -> "ID"
> 
>     2. "terraform" -> "Terraform"

done

**Comment by lvirbalas** [comment]
For reference: for those who have `protocol` not set from previous versions in the state file, this change will lead to a single occurrence of `protocol` field's value update from upper to lower case during the next `plan`?

**Comment by vbauzysvmware** [comment]
> For reference: for those who have `protocol` not set from previous versions in the state file, this change will lead to a single occurrence of `protocol` field's value update from upper to lower case during the next `plan`?

this happens today, cause vCD api returns lowercase and everytime update happens

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,7 +18,7 @@ FEATURES:
 * `resource/vcd_org_vdc` added Org VDC update and full state read - [GH-275]
 * `resource/vcd_org_vdc` added Org VDC metadata support - [GH-276]
 * `resource/vcd_snat` added ability to choose network name and type. [GH-282] 
-* `resource/vcd_dnat` added ability to choose network name and type. [GH-282]
+* `resource/vcd_dnat` added ability to choose network name and type. [GH-282] [GH-292] [GH-293]
```
Comment (Didainius) [inline CHANGELOG.md]: Never tested before, but should be like that as per https://www.terraform.io/docs/extend/best-practices/versioning.html
> IMPROVEMENTS:
> * resource/load_balancer: Add `ATTRIBUTE` argument (support X new functionality) [GH-12]
> * resource/subnet: Now better [GH-22, GH-32]

```suggestion
* `resource/vcd_dnat` added ability to choose network name and type. [GH-282, GH-292, GH-293]
```

Comment (vbauzysvmware) [reply CHANGELOG.md]: changed

File: website/docs/r/dnat.html.markdown [hunk 4]
```diff
@@ -52,9 +54,10 @@ The following arguments are supported:
...
```
Comment (lvirbalas) [inline website/docs/r/dnat.html.markdown]: I used org user and hit this with `ext`:
```
vcd_dnat.dnat_ext_net: Creating...

Error: unable to find orgVdcNetwork: my-org-network, err: can't find VDC Network: my-org-network
```
Let's add a sentence:
```
* `network_type` - (Optional; *v2.4+*) Type of the network on which to apply the NAT rule. Possible values `org` or `ext`. `ext` requires system administrator privileges. *`network_type` will be a required field in the next major version.*
```

**Pull request merged**